okhttp-tests/src/test/java/okhttp3/CipherSuiteTest.java

```diff
@@ -0,0 +1,116 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3;
+
+import org.junit.Test;
+
+import java.util.Arrays;
+
+import static okhttp3.CipherSuite.TLS_KRB5_WITH_DES_CBC_MD5;
+import static okhttp3.CipherSuite.TLS_RSA_EXPORT_WITH_RC4_40_MD5;
+import static okhttp3.CipherSuite.TLS_RSA_WITH_AES_128_CBC_SHA256;
+import static okhttp3.CipherSuite.forJavaName;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.fail;
+
+public class CipherSuiteTest {
+  @Test public void nullCipherName() {
+    try {
+      forJavaName(null);
+      fail("Should have thrown");
+    } catch (NullPointerException expected) {
+      // expected
+    }
+  }
+
+  @Test public void hashCode_usesIdentityHashCode() {
+    for (CipherSuite cipherSuite : Arrays.asList(
+        TLS_RSA_EXPORT_WITH_RC4_40_MD5,
+        TLS_RSA_WITH_AES_128_CBC_SHA256
+    )) {
+      assertEquals(cipherSuite.toString(),
+          System.identityHashCode(cipherSuite), cipherSuite.hashCode());
+    }
+  }
+
+  @Test public void instancesAreInterned() {
+    assertSame(forJavaName("TestCipherSuite"), forJavaName("TestCipherSuite"));
+    assertSame(CipherSuite.TLS_KRB5_WITH_DES_CBC_MD5,
+        forJavaName(TLS_KRB5_WITH_DES_CBC_MD5.javaName()));
+  }
+
+  @Test public void equals() {
+    assertEquals(forJavaName("cipher"), forJavaName("cipher"));
+    assertNotEquals(forJavaName("cipherA"), forJavaName("cipherB"));
+  }
+
+  @Test public void forJavaName_acceptsArbitraryStrings() {
+    // shouldn't throw
+    forJavaName("example CipherSuite name that is not in the whitelist");
+  }
+
+  @Test public void javaName_examples() {
+    assertEquals("SSL_RSA_EXPORT_WITH_RC4_40_MD5", TLS_RSA_EXPORT_WITH_RC4_40_MD5.javaName());
+    assertEquals("TLS_RSA_WITH_AES_128_CBC_SHA256", TLS_RSA_WITH_AES_128_CBC_SHA256.javaName());
+    assertEquals("TestCipherSuite", forJavaName("TestCipherSuite").javaName());
+  }
+
+  @Test public void javaName_equalsToString() {
+    for (CipherSuite cipherSuite : Arrays.asList(
+        TLS_RSA_EXPORT_WITH_RC4_40_MD5,
+        TLS_RSA_WITH_AES_128_CBC_SHA256
+    )) {
+      assertEquals(cipherSuite.javaName, cipherSuite.toString());
+    }
+  }
+
+  /**
+   * Tests backwards compatibility for OkHttp 3.2's behavior of allowing the prefix
+   * "TLS_" as an alias for 19 specific "SSL_" instances.
+   */
+  @Test public void forJavaName_fromLegacyEnumName() {
```

Comment:
Removed the backwards-compatibility support for the 19 aliases that used to be supported on previous versions of OkHttp. Adjusted the test.